bugsnag-plugin-android-ndk/src/main/jni/utils/serializer.c

```diff
@@ -694,20 +694,32 @@ void bsg_serialize_error(bsg_error exc, JSON_Object *exception,
   json_object_set_string(exception, "errorClass", exc.errorClass);
   json_object_set_string(exception, "message", exc.errorMessage);
   json_object_set_string(exception, "type", "c");
-  for (int findex = 0; findex < exc.frame_count; findex++) {
+  // assuming that the initial frame is the program counter. This logic will
+  // need to be revisited if (for example) we add more intelligent processing
+  // for stack overflow-type errors, like discarding the top frames, which
+  // would mean no stored frame is the program counter.
```

Comment:
Is there a ticket that tracks future improvements to this functionality?

Reply:
No, and no plans to do so. Just noted in here in case we ever do it.